src/components/dashboard/enrollments/ViewEnrollment.tsx

```diff
@@ -1,38 +1,66 @@
 import { Grid, Paper, Stack, Typography, Button } from '@mui/material';
-import { useLocation, useOutletContext, useParams } from 'react-router-dom';
+import {
+  useLocation,
+  useParams,
+  generatePath,
+  Link as RouterLink,
+} from 'react-router-dom';
+
+import HouseholdMemberTable from './HouseholdMemberTable';
 
 import Breadcrumbs from '@/components/elements/Breadcrumbs';
+import Loading from '@/components/elements/Loading';
+import { enrollmentName } from '@/modules/hmis/hmisUtil';
+import apolloClient from '@/providers/apolloClient';
 import { DashboardRoutes } from '@/routes/routes';
-import { Client } from '@/types/gqlTypes';
+import {
+  EnrollmentFieldsFragmentDoc,
+  useGetEnrollmentQuery,
+} from '@/types/gqlTypes';
 
 const ViewEnrollment = () => {
   const { pathname } = useLocation();
-  const { enrollmentId } = useParams() as {
+  const { clientId, enrollmentId } = useParams() as {
     enrollmentId: string;
+    clientId: string;
   };
-  const { client } = useOutletContext<{ client: Client | null }>();
-  if (!client) throw Error('Missing client');
+
+  const enrollmentFragment = apolloClient.readFragment({
+    id: `Enrollment:${enrollmentId}`,
+    fragment: EnrollmentFieldsFragmentDoc,
+  });
+
+  const { data, loading, error } = useGetEnrollmentQuery({
+    variables: { id: enrollmentId },
+    skip: !!enrollmentFragment,
+  });
+  if (error) throw error;
+  if (loading) return <Loading />;
+
+  const enrollment = enrollmentFragment || data?.enrollment;
+  if (!enrollment) throw Error('Enrollment not found');
```

Comment:
Seems like it will be a common pattern, see if there's a better way to do this using the cache configuration, or if it should be pulled into a custom hook

Reply:
@ZachCMP here I'm using `readFragment` to grab the enrollment if we have it in the cache, otherwise doing a query. On the previous page we queried for a list of enrollments (so we have `Enrollment:<id>` fragments in the cache). On this page, we should be able to use that cached enrollment to show the name of the enrollment at the top of the page. This seems like it will be a common pattern, but I'm wondering if we can configure to the cache to do this automatically? What do you think?